### PR TITLE
ci: add Github Action to check license header

### PR DESCRIPTION
**Pull request by torives**
Adds a new CI workflow to check for the license header in source files.

The action creates warnings for the files not covered in the config file, which is not a big deal but quite annoying. Unfortunately, I couldn't find a way to suppress them, but I'm open to suggestions.

Here are examples of [failed](https://github.com/cartesi/rollups-node/actions/runs/6974420025) and [successful](https://github.com/cartesi/rollups-node/actions/runs/6974497672) runs.

### CODE REVIEW

**Review by gligneul — DISMISSED** [review]
Very nice! I'm worried about the maintainability of having various small workflows for small jobs. But I won't block your PR because of that.

**Comment by torives** [comment]
> Very nice! I'm worried about the maintainability of having various small workflows for small jobs. But I won't block your PR because of that.

The changelog and image cleanup workflows have specific triggers, so they need to stay separate. But this and the conventional commit check could run after a push, so I could put them both in the same workflow. What do you say?

**Comment by marcelstanley** [comment]
> The changelog and image cleanup workflows have specific triggers, so they need to stay separate. But this and the conventional commit check could run after a push, so I could put them both in the same workflow. What do you say?

[go-code-quality.yml](https://github.com/cartesi/rollups-node/blob/next/.github/workflows/go-code-quality.yml) seems to be a good candidate for that merge as well.

**Comment by torives** [comment]
> [go-code-quality.yml](https://github.com/cartesi/rollups-node/blob/next/.github/workflows/go-code-quality.yml) seems to be a good candidate for that merge as well.

I thought that wouldn't work, but since the Go linter and the license check steps will ignore files unrelated to their tasks, I guess there's no problem putting them all together. Thanks for the suggestion!

**Comment by gligneul** [comment]
Please rebase this on top of next

**Review by gligneul — APPROVED** [review]
Looks good!

File: .github/workflows/code-quality.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Assess code quality
```
Comment (marcelstanley) [inline .github/workflows/code-quality.yml:1]: Perhaps we could change the script and the name in the future. I don't have a suggestion right now. 😉

**Review by marcelstanley — APPROVED** [review]

**Pull request merged**